### PR TITLE
Resurrects Cassandra's DependencyStore

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This resurrects Cassandra's DependencyStore, and notes the constraints
of the design. Most importantly, calls to `storeDependencies` will
overwrite values on a day granularity. This is different than anorm, for
example, which doesn't bucket by day.

See #597

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
rebased, but no other progress

**Comment by eirslett** [comment]
Oh, sorry for the lack of response!
+1 LGTM!

**Comment by codefromthecrypt** [comment]
To whom it may concern. I looked into resurrecting this. The problem is that there is very little test coverage to say what "should" happen. For example, the tests are trivial and rely mostly on whatever algebird is supposed to do. Best first step would be to backfill tests for the working implementation (anormdb and/or zipkin-hbase), then try to make a cassandra impl that passes those tests.

**Comment by rtyler** [comment]
See also [Working Effectively With Legacy Code](http://objectmentor.com/resources/articles/WorkingEffectivelyWithLegacyCode.pdf) <_<

**Comment by codefromthecrypt** [comment]
PTAL!

File: zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraDependencyStore.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.twitter.zipkin.common.Dependencies
+import com.twitter.zipkin.conversions.thrift._
+import com.twitter.zipkin.storage.DependencyStore
+import com.twitter.zipkin.thriftscala.{Dependencies => ThriftDependencies}
+import org.twitter.zipkin.storage.cassandra.Repository
+
+import scala.collection.JavaConverters._
+
+/**
+ * This implementation of DependencyStore assumes that the job aggregating dependencies
```
Comment (codefromthecrypt) [inline zipkin-cassandra/src/main/scala/com/twitter/zipkin/storage/cassandra/CassandraDependencyStore.scala:14]: this thing was the reason it took me so long to finish this. eventually, I figured out why everything was flooring to a day!

**Comment by codefromthecrypt** [comment]
merging so I can release, which could help progress https://github.com/openzipkin/zipkin-dependencies-hadoop

**Pull request merged**